libshvchainpack/src/chainpack/rpcdriver.cpp

```diff
@@ -403,8 +404,10 @@ RpcValue RpcDriver::decodeData(Rpc::ProtocolType protocol_type, const std::strin
 	}
 	catch(ParseException &e) {
 		nError() << Rpc::protocolTypeToString(protocol_type) << "Decode data error:" << e.msg();
-		std::string data_piece = data.substr(static_cast<size_t>(e.pos() - 10*16), 20*16);
-		nError().nospace() << "Start offset: " << start_pos << " Data: from pos:" << (e.pos() - 10*16) << "\n" << shv::chainpack::Utils::hexDump(data_piece);
+		auto err_pos = std::min(static_cast<size_t>(e.pos()), data.size());
+		auto hexdump_start_pos = err_pos - std::min(err_pos, static_cast<size_t>(10*16));
+		std::string data_piece = data.substr(hexdump_start_pos, 20*16);
+		nError().nospace() << "Start offset: " << start_pos << " Data: from pos:" << hexdump_start_pos << "\n" << shv::chainpack::Utils::hexDump(data_piece);
```

Comment:
I suggest use `logRpcDataW()` also here, as it is used in rest of implementation to log this kind of error.

Reply:
Ok, the log level is changed for all the nError messages in this function.